pycbc/inference/sampler_emcee.py

```diff
@@ -0,0 +1,994 @@
+
+        Parameters
+        -----------
+        fp : InferenceFile
+            A file handler to an open inference file.
+        max_iterations : {None, int}
+            See `write_chain` for details.
+        """
+        lls = self.likelihood_stats
+        ntemps, nwalkers, niterations = lls.shape
+
+        group = fp.stats_group + '/{field}/temp{tk}/walker{wi}'
+
+        if max_iterations is not None and max_iterations < niterations:
+            raise IndexError("The provided max size is less than the "
+                "number of iterations")
+
+        # create indices for faster sub-looping
+        widx = numpy.arange(nwalkers)
+        tidx = numpy.arange(ntemps)
+
+        # loop over stats
+        for stat in lls.fieldnames:
+            arr = lls[stat]
+            # create an empty array if desired, in case this is the first time
+            # writing
+            if max_iterations is not None:
+                out = numpy.zeros(max_iterations, dtype=arr.dtype)
+            # loop over temps
+            for tk in tidx:
+                # loop over number of walkers
+                for wi in widx:
+                    dataset_name = group.format(field=stat, tk=tk, wi=wi)
+                    try:
+                        fp[dataset_name][:niterations] = arr[tk,wi,:]
+                    except KeyError:
+                        # dataset doesn't exist yet, see if a larger array is
+                        # desired
+                        if max_iterations is not None:
+                            out[:niterations] = arr[tk,wi,:]
+                            fp[dataset_name] = out
+                        else:
+                            fp[dataset_name] = arr[tk,wi,:]
+
+    def write_results(self, fp, max_iterations=None):
+        """Writes metadata, samples, lnpost, lnprior,  and acceptance fraction
+        to the given file. See the write function for each of those for
+        details.
+
+        Parameters
+        -----------
+        fp : InferenceFile
+            A file handler to an open inference file.
+        max_iterations : {None, int}
+            If results have not previously been written to the
+            file, new datasets will be created. By default, the size of these
+            datasets will be whatever the length of the sampler's chain is at
+            this point. If you intend to run more iterations in the future,
+            set this value to that size so that the array in the file will be
+            large enough to accomodate future data.
+        """
+        self.write_metadata(fp)
+        self.write_chain(fp, max_iterations=max_iterations)
+        self.write_likelihood_stats(fp, max_iterations=max_iterations)
+        self.write_acceptance_fraction(fp)
+
+    @staticmethod
+    def _read_fields(fp, fields_group, fields, array_class,
+            thin_start=None, thin_interval=None, thin_end=None, iteration=None,
+            temps=None, walkers=None, flatten=True):
```

Comment:
__[flake8]__
* [E128] continuation line under-indented for visual indent

<!-- Comment by [SideCI](https://sideci.com) -->